src/raftstore/store/store.rs

```diff
@@ -2729,4 +2748,219 @@ mod tests {
+        let raft_client = Arc::new(RwLock::new(RaftClient::new(
+            env,
+            Arc::new(cfg.server),
+            Arc::clone(&sec_mgr),
+        )));
+
+        let (_, resolver) = new_resolver(Arc::clone(&pd_client)).unwrap();
+        let trans = ServerTransport::new(
+            raft_client,
+            snap_worker.scheduler(),
+            router.clone(),
+            resolver,
+        );
+
+        Store {
+            cfg: Rc::new(cfg.raft_store),
+            store: metapb::Store::new(),
+            kv_engine: Arc::clone(&db),
+            raft_engine: Arc::clone(&db),
+            sendch: sendch,
+            significant_msg_receiver: receiver,
+            region_peers: HashMap::default(),
+            pending_raft_groups: HashSet::default(),
+            split_check_worker: Worker::new("split check worker"),
+            region_worker: Worker::new("region worker"),
+            raftlog_gc_worker: Worker::new("raftlog gc worker"),
+            compact_worker: Worker::new("compact worker"),
+            pd_worker: FutureWorker::new("pd worker"),
+            consistency_check_worker: Worker::new("consistency check worker"),
+            apply_worker: Worker::new("apply worker"),
+            apply_res_receiver: None,
+            region_ranges: BTreeMap::new(),
+            pending_snapshot_regions: vec![],
+            trans: trans,
+            pd_client: pd_client,
+            coprocessor_host: coprocessor,
+            snap_mgr: snap_mgr,
+            raft_metrics: RaftMetrics::default(),
+            entry_cache_metries: Rc::new(RefCell::new(CacheQueryStats::default())),
+            pending_votes: RingQueue::with_capacity(PENDING_VOTES_CAP),
+            tag: "raftstore".to_owned(),
+            start_time: time::get_time(),
+            is_busy: false,
+            store_stat: StoreStat::default(),
+        }
+    }
+
+    fn new_mvcc_key(i: u8) -> Vec<u8> {
+        MvccKey::from_raw(&[i]).append_ts(0).encoded().to_owned()
+    }
+
+    // Initialize db with `n` SST files
+    fn init_db_with_sst_files<P: AsRef<Path>>(path: P, n: u8) -> DB {
+        let cfg = DbConfig::default();
+        let db_opts = cfg.build_opt();
+        let mut cfs_opts = cfg.build_cf_opts();
+        for opts in &mut cfs_opts {
+            opts.set_level_zero_file_num_compaction_trigger(1);
+        }
+
+        let db = new_engine_opt(path.as_ref().to_str().unwrap(), db_opts, cfs_opts).unwrap();
+
+        for cf in db.cf_names() {
+            let handle = db.cf_handle(cf).unwrap();
+            // Each SST file has only one kv.
+            for i in 0..n {
+                let k = keys::data_key(&new_mvcc_key(i));
+                db.put_cf(handle, &k, &[i]).unwrap();
+                db.flush_cf(handle, true).unwrap();
+            }
+        }
+
+        // Sleep 1s for compaction.
+        thread::sleep(Duration::from_secs(1));
+
+        db
+    }
+
+    #[test]
+    fn test_clear_stale_data() {
```

Comment:
Can you test it in integration tests?

Reply:
There is a raftstore node in integration tests already. You can just new a single node cluster and do whatever you want.